drv/i2c-api/src/lib.rs

```diff
@@ -510,6 +512,91 @@ impl I2cDevice {
         }
     }
 
+    ///
+    /// Writes a buffer, and then performs a subsequent register read.  These
+    /// are not performed as a single I2C transaction (that is, it is not a
+    /// repeated start) -- but the effect is the same in that the server does
+    /// these operations without an intervening receive (assuring that the
+    /// write can modify device state that the subsequent register read can
+    /// assume).
+    ///
+    pub fn write_read_reg<R: AsBytes, V: AsBytes + FromBytes>(
+        &self,
+        reg: R,
+        buffer: &[u8],
+    ) -> Result<V, ResponseCode> {
+        let mut val = V::new_zeroed();
+        let mut response = 0_usize;
+        let empty = [0u8; 1];
+
+        let (code, _) = sys_send(
+            self.task,
+            Op::WriteRead as u16,
+            &Marshal::marshal(&(
+                self.address,
+                self.controller,
+                self.port,
+                self.segment,
+            )),
+            response.as_bytes_mut(),
+            &[
+                Lease::from(buffer),
+                Lease::from(&empty[0..0]),
```

Comment:
It would be nice to be able to write
```suggestion
                Lease::from(&[]),
```

To make that work, [insert this code right here](https://github.com/oxidecomputer/hubris/blob/master/sys/userlib/src/lib.rs#L94)
```rust
impl<'a> From<&'a [u8; 0]> for Lease<'a> {
    fn from(x: &'a [u8; 0]) -> Self {
        Self::read_only(x)
    }
}
```

Reply:
Alternatively,
```suggestion
                Lease::read_only(&[]),
```
works already, without need to add a new `impl From`

Reply:
I love that alternative -- will change this throughout that file.